client/src/utils/api-util.js

```diff
@@ -1,107 +1,105 @@
-import * as AWS from "aws-sdk";
+import * as AWS from 'aws-sdk';
 
-// Constant for the S3 Bucket name where the Monument images are stored
-const s3ImageBucketName = 'monument-images';
+// Constant for the S3 Bucket name for the project
+const s3ImageBucketName = 'monuments-and-memorials';
 // Constant for the S3 folder name where the Monument images are stored
 const s3ImageBucketFolderName = 'images/';
 
+const httpMethodTypes = ['GET', 'POST', 'PUT'];
+
 /**
- * Send a GET request to the specified url
+ * Send a GET request to the specified URL
  * @param url - URL to send the GET to
  */
 export async function get(url) {
-    let error = null;
-    let res = await fetch(url)
-        .then(async (res) => {
-            if (!res.ok) {
-                let errorMessage = await res.text();
-
-                try {
-                    errorMessage = JSON.parse(errorMessage);
-                } catch (error) {}
-
-                throw Error(errorMessage.message);
-            }
-            return res;
-        })
-        .then(res => res.json())
-        .catch(err => error = err);
-    if (error || res.error) throw(error || res.error);
-    else return res;
+    return await sendRequest(url);
 }
 
 /**
- * Send a POST request to the specified url with the specified data
+ * Send a POST request to the specified URL with the specified data
  * @param url - URL to send the POST to
- * @param data - JSON data to send to the specified URL
- * @param contentType - The ContentType header, defaults to 'application/json'
+ * @param data - Data to send to the specified URL
  */
-export async function post(url, data, contentType = 'application/json') {
-    let error = null;
-    let res = await fetch(url, {
-        method: 'POST',
-        body: JSON.stringify(data),
-        headers: {
-            'Content-Type': contentType
-        }
-    })
-        .then(async (res) => {
-            if (!res.ok) {
-                let errorMessage = await res.text();
-
-                try {
-                    errorMessage = JSON.parse(errorMessage);
-                } catch (error) {}
-
-                throw Error(errorMessage.message);
-            }
-            return res;
-        })
-        .then(res => res.json())
-        .catch(err => error = err);
-    if (error || res.error) throw(error || res.error);
-    else return res;
+export async function post(url, data) {
```

Comment:
I don't know why I didn't recommend this earlier, but we've basically just recreated [https://www.npmjs.com/package/request-promise-native](request-promise-native) so I'm going to refactor everything to use that

Reply:
never mind I forgot how infuriating it is that this package can't use relative paths, so we'd have to store an environment variable with the server's domain name and I ain't about that

Reply:
We still need to accept headers on these since we don't use JSON in all our requests

Reply:
I added a `contentType` param to `post()`. There was already one on `put()`. I think we're good now, right?